controllers/front/listing/CategoryController.php

```diff
@@ -90,7 +90,7 @@ public function init()
         if (!Validate::isLoadedObject($this->category) || !$this->category->active || !$this->category->existsInShop($this->context->shop->id)) {
             if (!$this->category->id_type_redirected) {
                 if (in_array($this->category->redirect_type, [RedirectType::TYPE_CATEGORY_PERMANENT, RedirectType::TYPE_CATEGORY_TEMPORARY])) {
-                    $this->category->id_type_redirected = $this->category->id_category_default;
+                    $this->category->id_type_redirected = Category::getRootCategory()->id;
```

Comment:
Maybe I am wrong, but make sure to test this works.

The root category (ID 1) is NOT accessible in front office.
The main home category is **PS_HOME_CATEGORY** (ID 2 on a clean install).

Reply:
I did the test with breakpoints. This returns ID 2 on a fresh install (ID corresponding to is_root_category in database) in the front controller

Reply:
Cool, great then. I didn't know if it wouldn't return 1, the "root" thing in the method name is a bit confusing. :-)